### PR TITLE
Testnet tools update Nov 2022

### PR DESCRIPTION
**Pull request by dg-concordium**
## Purpose

Inform users about updates of cargo and client tools for P5.

## Changes

Downloads and release notes created.

## Checklist

- [X] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [ ] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [ ] (If necessary) I have updated the CHANGELOG.

## CLA acceptance

_Remove if not applicable.

By submitting the contribution I accept the terms and conditions of the
Contributor License Agreement v1.0
- link: https://developers.concordium.com/CLAs/Contributor-License-Agreement-v1.0.pdf

- [x] I accept the above linked CLA.

### CODE REVIEW

File: source/mainnet/net/installation/downloads-testnet.rst [hunk 1]
```diff
@@ -100,14 +100,12 @@ The version of the Ledger App is the same for Mainnet and Testnet. So if you alr
 
 .. _concordium-node-and-client-download-testnet:
 
-Concordium Client v4.2.0
+Concordium Client v5.0.0
 ========================
 
--  `Download the Testnet Concordium Client for Linux <https://distribution.concordium.software/tools/linux/concordium-client_4.2.0-0>`_
+-  `Download the Testnet Concordium Client for Linux <https://distribution.concordium.software/tools/linux/concordium-client_5.0.0-0>`_
```
Comment (abizjak) [inline source/mainnet/net/installation/downloads-testnet.rst:106]:
```suggestion
-  `Download the Testnet Concordium Client for Linux <https://distribution.concordium.software/tools/linux/concordium-client_5.0.1-0>`_
```

File: source/mainnet/net/installation/downloads-testnet.rst [hunk 2]
```diff
@@ -117,24 +115,24 @@ Concordium Client v4.2.0
 
    where you replace `*` with |client-version|.
 
--  `Download the Testnet Concordium Client for macOS <https://distribution.concordium.software/tools/macos/signed/concordium-client-4.2.0-0.pkg>`_
+-  `Download the Testnet Concordium Client for macOS <https://distribution.concordium.software/tools/macos/signed/concordium-client-5.0.0-0.pkg>`_
```
Comment (abizjak) [inline source/mainnet/net/installation/downloads-testnet.rst:118]:
```suggestion
-  `Download the Testnet Concordium Client for macOS <https://distribution.concordium.software/tools/macos/signed/concordium-client-5.0.1-0.pkg>`_
```

File: source/mainnet/net/installation/downloads-testnet.rst [hunk 3]
```diff
@@ -117,24 +115,24 @@ Concordium Client v4.2.0
...
 
    - The macOS distribution is an installer that places an alias to the binary
      into the folder ``/usr/local/bin``. So after installing, you should have
      ``concordium-client`` on your path.
 
--  `Download the Testnet Concordium Client for Windows <https://distribution.concordium.software/tools/windows/signed/concordium-client_4.2.0-0.exe>`_
+-  `Download the Testnet Concordium Client for Windows <https://distribution.concordium.software/tools/windows/signed/concordium-client_5.0.0-0.exe>`_
```
Comment (abizjak) [inline source/mainnet/net/installation/downloads-testnet.rst:124]:
```suggestion
-  `Download the Testnet Concordium Client for Windows <https://distribution.concordium.software/tools/windows/signed/concordium-client_5.0.1-0.exe>`_
```

File: source/mainnet/net/installation/downloads.rst [hunk 4]
```diff
@@ -100,10 +100,10 @@ Concordium Ledger App
 
 .. _concordium-node-and-client-download:
 
-Concordium Client v4.2.0
+Concordium Client v5.0.0
 ========================
 
--  `Download the Mainnet Concordium Client for Linux <https://distribution.concordium.software/tools/linux/concordium-client_4.2.0-0>`_
+-  `Download the Mainnet Concordium Client for Linux <https://distribution.concordium.software/tools/linux/concordium-client_5.0.0-0>`_
```
Comment (abizjak) [inline source/mainnet/net/installation/downloads.rst:106]:
```suggestion
-  `Download the Mainnet Concordium Client for Linux <https://distribution.concordium.software/tools/linux/concordium-client_5.0.1-0>`_
```

File: source/mainnet/net/installation/downloads.rst [hunk 5]
```diff
@@ -116,24 +116,24 @@ Concordium Client v4.2.0
 
    where you replace `*` with |client-version|.
 
--  `Download the Mainnet Concordium Client for macOS <https://distribution.concordium.software/tools/macos/signed/concordium-client-4.2.0-0.pkg>`_
+-  `Download the Mainnet Concordium Client for macOS <https://distribution.concordium.software/tools/macos/signed/concordium-client-5.0.0-0.pkg>`_
```
Comment (abizjak) [inline source/mainnet/net/installation/downloads.rst:119]:
```suggestion
-  `Download the Mainnet Concordium Client for macOS <https://distribution.concordium.software/tools/macos/signed/concordium-client-5.0.1-0.pkg>`_
```

File: source/mainnet/net/installation/downloads.rst [hunk 6]
```diff
@@ -116,24 +116,24 @@ Concordium Client v4.2.0
...
 
    - The macOS distribution is an installer that places an alias to the binary
      into the folder ``/usr/local/bin``. So after installing, you should have
      ``concordium-client`` on your path.
 
--  `Download the Mainnet Concordium Client for Windows <https://distribution.concordium.software/tools/windows/signed/concordium-client_4.2.0-0.exe>`_
+-  `Download the Mainnet Concordium Client for Windows <https://distribution.concordium.software/tools/windows/signed/concordium-client_5.0.0-0.exe>`_
```
Comment (abizjak) [inline source/mainnet/net/installation/downloads.rst:125]:
```suggestion
-  `Download the Mainnet Concordium Client for Windows <https://distribution.concordium.software/tools/windows/signed/concordium-client_5.0.1-0.exe>`_
```

File: source/mainnet/net/resources/release-notes-mainnet.rst [hunk 7]
```diff
@@ -12,6 +12,23 @@ Mainnet 4: Sirius
...
+
+- Build and test contracts using new protocol 5 features, such as upgradability and chain queries.
+- Support for relaxed smart contract resource restrictions in ``cargo concordium run``.
+- ``cargo concordium build`` now checks contracts with respect to protocol version 5 semantics.
+
+Concordium Client 5.0.0
+-----------------------
+
+Concordium Client 5.0.0 adds support for the upcoming `protocol version 5 <https://github.com/Concordium/concordium-update-proposals/blob/main/updates/P5.txt>`_ which is planned for release on Testnet November 17, 2022.
+It also adds a ``--secure`` flag to enable connecting to gRPC using TLS. All commands that query the node support this.
```
Comment (abizjak) [inline source/mainnet/net/resources/release-notes-mainnet.rst:30]: In 5.0.1, which we are releasing, there are also these changes

```
- Add support of contract schema V3.
	  - V3 schemas offer the same options as V2, but also optionally includes a schema for contract events.
	  - `transaction status` now displays contract events, and a schema can be provided with `--schema`, which 
	    will be used to parse the contract events. By default events are parsed with the schema embedded in the
      contract, if present.
	  - This enables concordium-client to interact with contracts and schemas 
	    using `concordium-std` version 5.
- Improved formatting of `transaction status` output using contract schemas if
  they are available for displaying contract events.
- Output function parameters as hex strings in `transaction status`.
```

File: source/mainnet/net/resources/release-notes.rst [hunk 8]
```diff
@@ -12,6 +12,23 @@ Sirius Testnet
...
+
+- Build and test contracts using new protocol 5 features, such as upgradability and chain queries.
+- Support for relaxed smart contract resource restrictions in ``cargo concordium run``.
+- ``cargo concordium build`` now checks contracts with respect to protocol version 5 semantics.
+
+Concordium Client 5.0.0
+-----------------------
+
+Concordium Client 5.0.0 adds support for the upcoming `protocol version 5 <https://github.com/Concordium/concordium-update-proposals/blob/main/updates/P5.txt>`_ which is planned for release on Testnet November 17, 2022.
+It also adds a ``--secure`` flag to enable connecting to gRPC using TLS. All commands that query the node support this.
```
Comment (abizjak) [inline source/mainnet/net/resources/release-notes.rst:30]: Same here.

File: source/mainnet/variables.rst [hunk 9]
```diff
@@ -12,9 +12,9 @@
 .. |bw| replace:: Concordium Wallet for Web
 
 .. Client verification variables
-.. |client-linux| replace:: concordium-client_4.2.0-0
-.. |client-version| replace:: 4.2.0-0
-.. |client-linux-checksum| replace:: 805bb509b1b76bd6750e744398146b064876e2251ed724bbcf26b229d6ede627
+.. |client-linux| replace:: concordium-client_5.0.0-0
+.. |client-version| replace:: 5.0.0-0
+.. |client-linux-checksum| replace:: a7bbba26b485f0841eefb5277cf8c4ec9044faa599251fca1bac16c0991392b3
```
Comment (abizjak) [inline source/mainnet/variables.rst:17]: This should be updated as well I think.

File: source/mainnet/variables.rst [hunk 10]
```diff
@@ -12,9 +12,9 @@
 .. |bw| replace:: Concordium Wallet for Web
 
 .. Client verification variables
-.. |client-linux| replace:: concordium-client_4.2.0-0
-.. |client-version| replace:: 4.2.0-0
-.. |client-linux-checksum| replace:: 805bb509b1b76bd6750e744398146b064876e2251ed724bbcf26b229d6ede627
+.. |client-linux| replace:: concordium-client_5.0.0-0
+.. |client-version| replace:: 5.0.0-0
```
Comment (abizjak) [inline source/mainnet/variables.rst:16]:
```suggestion
.. |client-version| replace:: 5.0.1-0
```

File: source/mainnet/variables.rst [hunk 11]
```diff
@@ -12,9 +12,9 @@
 .. |bw| replace:: Concordium Wallet for Web
 
 .. Client verification variables
-.. |client-linux| replace:: concordium-client_4.2.0-0
-.. |client-version| replace:: 4.2.0-0
-.. |client-linux-checksum| replace:: 805bb509b1b76bd6750e744398146b064876e2251ed724bbcf26b229d6ede627
+.. |client-linux| replace:: concordium-client_5.0.0-0
```
Comment (abizjak) [inline source/mainnet/variables.rst:15]:
```suggestion
.. |client-linux| replace:: concordium-client_5.0.1-0
```

**Review by abizjak — COMMENTED** [review]
We are releasing 5.0.1 of the concordium-client.

It additionally has support for V3 schemas, the same as cargo-concordium 2.4.0 has.

**Review by mh-concordium — APPROVED** [review]
Looks good to me.

File: source/mainnet/net/installation/downloads.rst [hunk 12]
```diff
@@ -116,24 +116,24 @@ Concordium Client v4.2.0
...
 Download cargo-concordium:
 
-   -  `Download Mainnet cargo-concordium for Linux <https://distribution.concordium.software/tools/linux/cargo-concordium_2.2.0>`_
+   -  `Download Mainnet cargo-concordium for Linux <https://distribution.concordium.software/tools/linux/cargo-concordium_2.4.0>`_
 
-   -  `Download Mainnet cargo-concordium for MacOS <https://distribution.concordium.software/tools/macos/cargo-concordium_2.2.0>`_
+   -  `Download Mainnet cargo-concordium for MacOS <https://distribution.concordium.software/tools/macos/cargo-concordium_2.4.0>`_
 
-   -  `Download Mainnet cargo-concordium for Windows <https://distribution.concordium.software/tools/windows/cargo-concordium_2.2.0.exe>`_
+   -  `Download Mainnet cargo-concordium for Windows <https://distribution.concordium.software/tools/windows/cargo-concordium_2.4.0.exe>`_
```
Comment (abizjak) [inline source/mainnet/net/installation/downloads.rst:136]:
```suggestion
   -  `Download Mainnet cargo-concordium for Windows <https://distribution.concordium.software/tools/windows/signed/cargo-concordium_2.4.0.exe>`_
```

**Review by abizjak — APPROVED** [review]

File: source/mainnet/net/installation/downloads-testnet.rst [hunk 13]
```diff
@@ -117,24 +115,24 @@ Concordium Client v4.2.0
...
+Cargo-concordium v2.4.0
 =======================
 
 Download cargo-concordium:
 
-   -  `Download Testnet cargo-concordium for Linux <https://distribution.concordium.software/tools/linux/cargo-concordium_2.2.0>`_
+   -  `Download Testnet cargo-concordium for Linux <https://distribution.concordium.software/tools/linux/cargo-concordium_2.4.0>`_
 
-   -  `Download Testnet cargo-concordium for MacOS <https://distribution.concordium.software/tools/macos/cargo-concordium_2.2.0>`_
+   -  `Download Testnet cargo-concordium for MacOS <https://distribution.concordium.software/tools/macos/cargo-concordium_2.4.0>`_
```
Comment (abizjak) [inline source/mainnet/net/installation/downloads-testnet.rst:133]:
```suggestion
   -  `Download Testnet cargo-concordium for MacOS <https://distribution.concordium.software/tools/macos/signed/cargo-concordium_2.4.0>`_
```

File: source/mainnet/net/installation/downloads.rst [hunk 14]
```diff
@@ -116,24 +116,24 @@ Concordium Client v4.2.0
...
+Cargo-concordium v2.4.0
 =======================
 
 Download cargo-concordium:
 
-   -  `Download Mainnet cargo-concordium for Linux <https://distribution.concordium.software/tools/linux/cargo-concordium_2.2.0>`_
+   -  `Download Mainnet cargo-concordium for Linux <https://distribution.concordium.software/tools/linux/cargo-concordium_2.4.0>`_
 
-   -  `Download Mainnet cargo-concordium for MacOS <https://distribution.concordium.software/tools/macos/cargo-concordium_2.2.0>`_
+   -  `Download Mainnet cargo-concordium for MacOS <https://distribution.concordium.software/tools/macos/cargo-concordium_2.4.0>`_
```
Comment (abizjak) [inline source/mainnet/net/installation/downloads.rst:134]:
```suggestion
   -  `Download Mainnet cargo-concordium for MacOS <https://distribution.concordium.software/tools/macos/signed/cargo-concordium_2.4.0>`_
```

**Review by abizjak — APPROVED** [review]

**Pull request merged**